bittensor/_dataset/dataset_impl.py

```diff
@@ -1,536 +1,490 @@
-        else:
-            logger.warning("Failed to get text".ljust(20) + "<blue>{}</blue>".format(file_meta['Name']))
-            self.IPFS_fails += 1
-            
-        return text 
+        self.dataset_size_map = {}
+        self.dataset_hash_map = dataset_hash_map
+        for  k,file_meta_list in dataset_hash_map.items():
+            all_text_file_metas += v
+            self.dataset_size_map[k] =  sum([f['Size'] for f in file_meta_list])
+            self.dataset_size += self.dataset_size_map[k]
+        self.all_text_file_metas = all_text_file_metas
+
+    construct_text_corpus = build_datasets
+    async def async_save_json(self, 
+                              path:str,
+                              obj:Union[dict, list],
+                              include_root:bool=True) -> str:
+        """ 
+        Async save of json for storing text hashes
 
-    def get_dataset(self , file_meta):
-        r""" Either load a dataset, which is a list of hashes, from disk or download it from IPFS
         Args:
-            file_meta (dict of str: int):
-                Specify the details of the dataset in the format of {'Name': , 'Hash':}.
-
-        Return:
-            hashes (list):
-                The hashes from the dataset downloaded from disk or IPFS.     
+            path (List[str]):
+                Axon to serve.
+            obj (bool):
+                The object to save locally
+            include_root (bool):
+                Include self.data_dir as the prefix.
+                    - if True, ths meants shortens the batch and 
+                    specializes it to be with respect to the dataset's 
+                    root path which is in ./bittensor/dataset
+            
+        Returns: 
+            path (str)
+                Path of the saved JSON
         """
-        # --- Load text from path
-        logger.success( f"Getting dataset: {file_meta['Name']}" )
         
-        hashes = self.load_hash(file_meta)
+        if include_root:
+            path = os.path.join(self.data_dir, path)
 
-        if hashes != None:
-            hashes = json.loads(hashes)
+        dir_path = os.path.dirname(path)
 
-        # --- If couldnt load from path, download text.
-        else:
-            response = self.get_ipfs_directory(self.dataset_dir, file_meta)
-            if (response != None) and (response.status_code == 200):
-                self.IPFS_fails = 0
-                hashes = response.json()
-
-                # --- Save text if the save_dataset flag is on.
-                if self.save_dataset :
-                    self.save_hash(file_meta, json.dumps(response.json()) )
-                    
-            else:
-                self.IPFS_fails += 1
-                logger.warning("Failed to get dataset".ljust(20) + "<blue>{}</blue>".format(file_meta['Name']))
-                return None
+        # ensure the json is the prefix
+        if path[-len('.json'):] != '.json':
+            path += '.json'
```

Comment:
fyi, https://docs.python.org/3/library/os.path.html#os.path.splitext is easier/more legibile for the future